autocomplete-input-text.js

```diff
@@ -1,128 +1,246 @@
-			dropdownLabel: {
-				type: String,
-				value: null,
-			},
-			maxHeight: {
-				type: Number,
-			},
-			minLength: {
-				type: String,
-			},
-			remoteSource: {
-				type: Boolean
-			},
-			selectFirst: {
-				type: Boolean,
-			},
-			showOnFocus: {
-				type: Boolean,
-			},
+			data: { type:Object },
+			dropdownLabel: { type: String, attribute: 'dropdown-label' },
+			maxLength: { type: String, attribute: 'max-length' },
+			minLength: { type: String, attribute: 'min-length' },
+			remoteSource: { type: Boolean, attribute: 'remote-source' },
+			showOnFocus: { type: Boolean, attribute: 'show-on-focus' },
+			_showSuggestions: { state: true },
+			_minWidth: { state: true },
+			_filter: { state: true },
+			_suggestions: { state: true },
 			/**
 			* These properties are used by d2l-input-text
 			*/
-			ariaLabel: {
-				type: String
-			},
-			maxLength: {
-				type: String,
-			},
-			placeholder: {
-				type: String,
-			},
-			type: {
-				type: String,
-				value: 'text'
-			},
-			value: {
-				type: String,
-			}
+			ariaLabel: { type: String, attribute: 'aria-label' },
+			placeholder: { type: String },
+
 		};
 	}
+	static get styles() {
+		return [bodyCompactStyles, css`
+			:host {
+				display: flex;
+			}
+
+			#d2l-labs-autocomplete-list {
+				margin: 0;
+				padding: 0;
+			}
+
+			#d2l-labs-autocomplete-dropdown-label {
+				font-size: 0.7rem;
+				padding: 0.4rem 0.7rem;
+			}
+
+			.d2l-labs-autocomplete-suggestion {
+				cursor: pointer;
+				margin-right: 0.1rem;
+				outline: none;
+				overflow-x: hidden;
+				padding: 0.4rem 0.7rem;
+				text-align: left;
+				text-overflow: ellipsis;
+				white-space: nowrap;
+			}
+
+			.d2l-labs-autocomplete-suggestion-highlighted {
+				font-weight: bold;
+			}
+
+			.d2l-labs-autocomplete-suggestion:focus,
+			.d2l-labs-autocomplete-suggestion:hover {
+				background-color: var(--d2l-color-celestine-plus-2);
+			}
+		`];
+	}
+
 	constructor() {
 		super();
+		this.data = [];
+		this.dropdownLabel = null;
+		this.minLength = 1;
+		this.remoteSource = false;
+		this.showOnFocus = false;
+		this._filter = '';
+		this._showSuggestions = false;
+		this._suggestions = [];
 		this._uniqueId = getUniqueId();
+
+	}
+
+	render() {
+		return html`<d2l-dropdown
+			@d2l-dropdown-close=${() => this._showSuggestions = false}
+			class="d2l-labs-autocomplete-dropdown-wrapper"
+			no-auto-open>
+			<div class="d2l-dropdown-opener">
+				<d2l-input-text
+						aria-label="${this.ariaLabel}"
+						@input=${this._onInput}
+						@keydown=${this._onKeyDown}
+						@focus=${this._onFocus}
+						maxlength="${ifDefined(this.maxLength)}"
+						novalidate
+						autocomplete="list"
+						placeholder="${ifDefined(this.placeholder)}"
+						role="combobox">
+				</d2l-input-text>
+			</div>
+			<d2l-dropdown-content
+				id="d2l-labs-autocomplete-dropdown-content"
+				no-auto-focus
+				max-height="${this.maxHeight}"
+				min-width="${this._minWidth}"
+				no-padding=""
+				no-pointer=""
```

Comment:
Boolean attributes don't have values -- this sometimes happened in Polymer due to an automatic migration.
```suggestion
				no-padding
				no-pointer
```